app/api/enrollments.js

```diff
@@ -0,0 +1,123 @@
+const router = require('express').Router({ mergeParams: true })
+const db = require('../models/index')
+const courseService = require('../services/course_service')
+const enrollmentService = require('../services/enrollment_service')
+const sectionService = require('../services/section_service')
+const { requireAuthentication } = require('../../lib/auth')
+const string_helpers = require('../../lib/string_helpers')
+const { logger } = require('../../lib/logger')
+
+// GET /courses/:course_id/enrollments
+// endpoint handles teacher getting a roster for a given course
+router.get('/', requireAuthentication, async function (req, res) {
+    const user = await db.User.findByPk(req.payload.sub) // find user by ID, which is stored in sub
+    const courseId = parseInt(req.params['course_id'])
+    // we want to list the roster only if the user for the course is a teacher
+    const enrollmentTeacher = await db.Enrollment.findOne({
+        where: { 
+            userId: user.id,
+            courseId: courseId,
+            role: 'teacher'
+        }
+    })
+
+    if (enrollmentTeacher) {
+        const enrollments = await db.Enrollment.findAll({
+            include: [
+                {
+                    model: db.Section,
+                    where: {
+                        courseId: courseId
+                    }
+                    
+                }
+            ]
+            
+        })
+        res.status(200).send({
+            enrollments: enrollmentService.extractArrayEnrollmentFields(enrollments)
+        })
+    } else {
+        res.status(403).send({error: `Only the teacher for a course can view the roster`})
+    }
+})
+
+// deletes a student from the course roster
+router.delete('/:enrollment_id', requireAuthentication, async function (req, res) {
+    const user = await db.User.findByPk(req.payload.sub) // find user by ID, which is stored in sub
+    const courseId = parseInt(req.params['course_id'])
+    const enrollmentId = parseInt(req.params['enrollment_id'])
+
+    // we want to delete an enrollment only if the user for the course is a teacher
+    const enrollmentTeacher = await db.Enrollment.findOne({
+        where: { 
+            userId: user.id,
+            courseId: courseId,
+            role: 'teacher'
+        }
+    })
+
+    if (enrollmentTeacher) {
+        const enrollment = await db.Enrollment.findByPk(enrollmentId)
```

Comment:
This would allow a teacher to delete any enrollment. This should validate that the enrollment is a student enrolled in the course for which this teacher is authenticated